### ISSUE TITLE
Breaking changes

### ISSUE THREAD

**Issue by rubensworks** [comunica/sparqlee#122]
@jitsedesmet It looks like something went wrong while checking whether or not changes in sparqlee caused breakages in Comunica. Everything from the master HEAD until [this commit](https://github.com/comunica/sparqlee/commit/b527ac2c9dc6677b06bb5cd75c8ee96539adb9d4) causes unit, spec and integration test failures.

I have no bandwidth to look into fixing all of the problems. So my current idea is to revert master back to the state [before the OverloadTree was added](https://github.com/comunica/sparqlee/commit/c4959d634995a1f5724d148b22880ed60b92da23), as everything still worked there, and there were no performance regressions yet. Unless you see any other solutions @jitsedesmet?

**Comment by jitsedesmet** [comment]
Oh, I'm really sorry to hear that. I'm out for the week so I think it would be best to revert the changes until then. I will take a closer look at the issues when I'm back. I must have done something wrong, sorry!

**Comment by jitsedesmet** [comment]
A does the mentioned commit work? It seems like it's the commit where we introduced the new cache. Maybe I needed to add this module dependency somewhere and didn't?

**Comment by rubensworks** [comment]
No, that one doesn't work. I'll just revert (as I have to prepare a new release ASAP)

**Comment by jitsedesmet** [comment]
With even more pain in my hearth I suggest rolling back further than this then, to the commit before we touched the type system since the new type system without a cache is way to slow.

**Comment by rubensworks** [comment]
Up until what commit would you suggest then?

**Comment by jitsedesmet** [comment]
Since I'm one my phone it's pretty hard to get the reference but here https://github.com/comunica/sparqlee/pull/112#discussion_r695577156 I mentioned a few commits, the commit with lowest execution time is the one we want. This is where we introduced the overloadTree but didn't touch the type system.

**Comment by jitsedesmet** [comment]
I think it was this commit https://github.com/jitsedesmet/sparqlee/commit/c4959d634995a1f5724d148b22880ed60b92da23. Again, I'm really sorry for all this.

**Comment by rubensworks** [comment]
Ah yeah, that's the one I was referring to in the issue description :-)
So I'll just revert to right before that one.

**Comment by rubensworks** [comment]
HEAD state at the time of writing is now in https://github.com/comunica/sparqlee/commits/reverted/jitse, and master has been reverted.
